src/boss/cloudvm/worker_config_template.go

```diff
@@ -1,42 +1,49 @@
 package cloudvm
 
 import (
-	"fmt"
-	"os"
+	"encoding/json"
+	"log"
 	"path/filepath"
 
 	"github.com/open-lambda/open-lambda/ol/common"
 )
 
-func LoadWorkerConfigTemplate(templatePath string, workerPath string) error {
-	// Create template.json if it doesn't exist
-	if _, err := os.Stat(templatePath); os.IsNotExist(err) {
-		// If template.json does not exist, load defaults
-		if err := common.LoadDefaults(workerPath); err != nil {
-			return fmt.Errorf("failed to load defaults: %v", err)
-		}
+func SaveTemplateConfToWorkerDir(cfg *common.Config, workerPath string, workerPort string) error {
+	// Copy the config so we can safely mutate it
+	cfgCopy := deepCopyConfig(cfg)
+	defaultCfg, _ := common.GetDefaultWorkerConfig(workerPath)
 
-		// Set the worker port
-		common.Conf.Worker_port = GetLocalPlatformConfigDefaults().Worker_Starting_Port
-
-		// Save the updated configuration to template.json
-		if err := common.SaveConf(templatePath); err != nil {
-			return fmt.Errorf("failed to save updated configuration to template.json: %v", err)
-		}
-	} else if err != nil {
-		return fmt.Errorf("error checking template.json: %v", err)
+	// Patch fields ONLY if they're empty
+	if cfgCopy.Worker_dir == "" {
+		cfgCopy.Worker_dir = defaultCfg.Worker_dir
+		log.Printf("Patched Worker_dir: %s", cfg.Worker_dir)
 	}
-
-	// load template.json
-	if err := common.LoadConf(templatePath); err != nil {
-		return fmt.Errorf("failed to load template.json: %v", err)
+	if cfgCopy.Registry == "" {
+		cfgCopy.Registry = defaultCfg.Registry
+		log.Printf("Patched Registry: %s", cfg.Registry)
+	}
+	if cfgCopy.Pkgs_dir == "" {
+		cfgCopy.Pkgs_dir = defaultCfg.Pkgs_dir
+		log.Printf("Patched Pkgs_dir: %s", cfg.Pkgs_dir)
+	}
+	if cfgCopy.SOCK_base_path == "" {
+		cfgCopy.SOCK_base_path = defaultCfg.SOCK_base_path
+		log.Printf("Patched SOCK_base_path: %s", cfg.SOCK_base_path)
 	}
+	if cfgCopy.Import_cache_tree == "" {
+		cfgCopy.Import_cache_tree = defaultCfg.Import_cache_tree
+		log.Printf("Patched Import_cache_tree: %s", cfg.Import_cache_tree)
+	}
+	cfgCopy.Worker_port = workerPort
 
 	// Save the template configuration to the worker's config directory
 	configPath := filepath.Join(workerPath, "config.json")
-	if err := common.SaveConf(configPath); err != nil {
-		return fmt.Errorf("failed to save updated configuration to worker config: %v", err)
-	}
+	return common.ExportConfig(cfgCopy, configPath)
+}
 
-	return nil
+func deepCopyConfig(src *common.Config) *common.Config {
```

Comment:
If we don't have pointers to sub structs, is there a need for a deep copy?  I guess there are the "any" fields, but we don't change those, do we?